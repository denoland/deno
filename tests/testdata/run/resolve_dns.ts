const nameServer = { nameServer: { ipAddr: "127.0.0.1", port: 4553 } };

const [a, aaaa, aname, caa, cname, mx, naptr, ns, ptr, soa, srv, txt] =
  await Promise
    .all([
      Deno.resolveDns("www.example.com", "A", nameServer),
      Deno.resolveDns("www.example.com", "AAAA", nameServer),
      Deno.resolveDns("www.example.com", "ANAME", nameServer),
      Deno.resolveDns("example.com", "CAA", nameServer),
      Deno.resolveDns("alias.example.com", "CNAME", nameServer),
      Deno.resolveDns("example.com", "MX", nameServer),
      Deno.resolveDns("example.com", "NAPTR", nameServer),
      Deno.resolveDns("example.com", "NS", nameServer),
      Deno.resolveDns("1.2.3.4.IN-ADDR.ARPA.", "PTR", nameServer),
      Deno.resolveDns("example.com", "SOA", nameServer),
      Deno.resolveDns("_service._tcp.example.com", "SRV", nameServer),
      Deno.resolveDns("example.com", "TXT", nameServer),
    ]);

console.log("A");
console.log(JSON.stringify(a));

console.log("AAAA");
console.log(JSON.stringify(aaaa));

console.log("ANAME");
console.log(JSON.stringify(aname));

console.log("CAA");
console.log(JSON.stringify(caa));

console.log("CNAME");
console.log(JSON.stringify(cname));

console.log("MX");
console.log(JSON.stringify(mx));

console.log("NAPTR");
console.log(JSON.stringify(naptr));

console.log("NS");
console.log(JSON.stringify(ns));

console.log("PTR");
console.log(JSON.stringify(ptr));

console.log("SOA");
console.log(JSON.stringify(soa));

console.log("SRV");
console.log(JSON.stringify(srv));

console.log("TXT");
console.log(JSON.stringify(txt));

try {
  await Deno.resolveDns("not-found-example.com", "A", nameServer);
} catch (e) {
  console.log(
    `Error ${
      e instanceof Error ? e.name : "[non-error]"
    } thrown for not-found-example.com`,
  );
}

try {
  // @ts-ignore testing invalid overloads
  await Deno.resolveDns("example.com", "SSHFP", nameServer);
} catch (e) {
  console.log((e as Error).message);
}

try {
  const ac = new AbortController();
  queueMicrotask(() => ac.abort());
  await Deno.resolveDns("www.example.com", "A", {
    ...nameServer,
    signal: ac.signal,
  });
} catch (e) {
  console.log((e as Error).name);
}

try {
  const ac = new AbortController();
  ac.abort();
  await Deno.resolveDns("www.example.com", "A", {
    ...nameServer,
    signal: ac.signal,
  });
} catch (e) {
  console.log((e as Error).name);
}
