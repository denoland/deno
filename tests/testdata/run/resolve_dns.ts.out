A
["1.2.3.4","5.6.7.8"]
AAAA
["1:2:3:4:5:6:7:8"]
ANAME
["aname.example.com."]
CAA
[{"critical":false,"tag":"issue","value":"ca.example.net"},{"critical":false,"tag":"issue","value":"ca2.example.net; account=123456"},{"critical":false,"tag":"issuewild","value":";"},{"critical":false,"tag":"iodef","value":"mailto:security@example.com"},{"critical":true,"tag":"tbs","value":"Unknown"}]
CNAME
["cname.example.com."]
MX
[{"preference":10,"exchange":"mx1.com."},{"preference":20,"exchange":"mx2.com."}]
NAPTR
[{"order":10,"preference":0,"flags":"s","services":"SIPS+D2T","regexp":"","replacement":"_sips._tcp.example.com."},{"order":10,"preference":0,"flags":"s","services":"RELAY:turn.udp","regexp":"","replacement":"_turn._udp.example.com."}]
NS
["ns1.ns.com.","ns2.ns.com.","ns3.ns.com."]
PTR
["www.example.com.","alias.example.com."]
SOA
[{"mname":"net.example.com.","rname":"admin\\.domain.example.com.","serial":20,"refresh":7200,"retry":600,"expire":3600000,"minimum":60}]
SRV
[{"priority":0,"weight":100,"port":1234,"target":"srv.example.com."}]
TXT
[["I","am","a","txt","record"],["I","am","another","txt","record"],["I am a different","txt record"],["key=val"]]
Error NotFound thrown for not-found-example.com
Provided record type is not supported
AbortError
AbortError
